api/v1alpha1/ratelimit_types.go

```diff
@@ -91,6 +92,62 @@ type RateLimitRule struct {
 	// 429 HTTP status code is sent back to the client when
 	// the selected requests have reached the limit.
 	Limit RateLimitValue `json:"limit"`
+	// RequestHitsAddend specifies the number to reduce the rate limit counters
+	// on the request path. If the addend is not specified, the default behavior
+	// is to reduce the rate limit counters by 1.
+	//
+	// When Envoy receives a request that matches the rule, it tries to reduce the
+	// rate limit counters by the specified number. If the counter doesn't have
+	// enough capacity, the request is rate limited.
+	//
+	// +optional
+	RequestHitsAddend *RateLimitHitsAddend `json:"requestHitsAddend,omitempty"`
+	// ResponseHitsAddend specifies the number to reduce the rate limit counters
+	// after the response is sent back to the client or the request stream is closed.
+	//
+	// The addend is used to reduce the rate limit counters for the matching requests.
+	// Since the reduction happens after the request stream is complete, the rate limit
+	// won't be enforced for the current request, but for the subsequent matching requests.
+	//
+	// This is optional and if not specified, the rate limit counters are not reduced
+	// on the response path.
+	//
+	// Currently, this is only supported for HTTP Global Rate Limits.
+	//
+	// +optional
+	ResponseHitsAddend *RateLimitHitsAddend `json:"responseHitsAddend,omitempty"`
```

Comment:
I guess RequestHitsAddend and ResponseHitsAddend can't coexist?

RequestHitsAddend and  ResponseHitsAddend should also be hidden?

Reply:
Well it can coexist

Reply:
cc @zirain has more contexts on how to implmente this

Reply:
it could be co-exist, e.g. count 1 on request, and the (real usage -1) on response, this will give us better control of limitation.